rust/transaction-filter/src/filters/event_filter.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{filters::MoveStructTagFilter, json_search::JsonSearchTerm, traits::Filterable};
+use anyhow::Error;
+use aptos_protos::transaction::v1::{move_type::Content, Event};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct EventFilter {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub data: Option<JsonSearchTerm>,
+    // Only for events that have a struct as their generic
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub struct_type: Option<MoveStructTagFilter>,
+    // TODO: address? etc
```

Comment:
Like the address that emitted the event for v1 events?

Reply:
yeah- I just removed this comment, don't think it's worth it